src/stlink-lib/usb.c

```diff
@@ -1236,15 +1270,12 @@ stlink_t *stlink_open_usb(enum ugly_loglevel verbose, int reset, char serial[STL
 
         if (ret) { continue; } // could not open device
 
-        sl->serial_size = libusb_get_string_descriptor_ascii(
-            handle, desc.iSerialNumber, (unsigned char *)sl->serial, sizeof(sl->serial));
+        ret = stlink_serial(handle, &desc, sl->serial);
 
         libusb_close(handle);
 
-        if (sl->serial_size < 0) { continue; } // could not read serial
```

Comment:
It would be nice to keep the check of getting serial number ( `if (ret < 0)...` )

Reply:
well spotted